src/main/cpp/hw/factories/DragonControlToCTREProAdapterFactory.cxx

```diff
@@ -14,46 +14,138 @@
+                                                                                   ctre::phoenixpro::hardware::TalonFX &controller)
 {
-public:
-    static DragonControlToCTREV5AdapterFactory *GetFactory();
-    DragonControlToCTREV5Adapter *CreateAdapter(std::string networkTableName,
-                                                int controllerSlot,
-                                                const ControlData &controlInfo,
-                                                const DistanceAngleCalcStruc &calcStruc,
-                                                ctre::phoenix::motorcontrol::can::WPI_BaseMotorController *controller);
-    DragonControlToCTREV5Adapter *CreatePercentOuptutAdapter(std::string networkTableName,
-                                                             ctre::phoenix::motorcontrol::can::WPI_BaseMotorController controller);
-
-private:
-    DragonControlToCTREV5AdapterFactory() = default;
-    ~DragonControlToCTREV5AdapterFactory() = default;
-
-    static DragonControlToCTREV5AdapterFactory *m_factory;
-};
\ No newline at end of file
+    switch (controlInfo.GetMode())
+    {
+    case ControlModes::CONTROL_TYPE::PERCENT_OUTPUT:
+        return new DragonPercentOutputToCTREProAdapter(networkTableName,
+                                                       controllerSlot,
+                                                       controlInfo,
+                                                       calcStruc,
+                                                       controller);
+        break;
+    case ControlModes::CONTROL_TYPE::POSITION_ABS_TICKS:
+        return new DragonTicksToCTREProAdapter(networkTableName,
+                                               controllerSlot,
+                                               controlInfo,
+                                               calcStruc,
+                                               controller);
+        break;
+
+    case ControlModes::CONTROL_TYPE::POSITION_DEGREES:
+        return new DragonPositionDegreeToCTREProAdapter(networkTableName,
+                                                        controllerSlot,
+                                                        controlInfo,
+                                                        calcStruc,
+                                                        controller);
+        break;
+
+    case ControlModes::CONTROL_TYPE::POSITION_INCH:
+        return new DragonPositionInchToCTREProAdapter(networkTableName,
+                                                      controllerSlot,
+                                                      controlInfo,
+                                                      calcStruc,
+                                                      controller);
+        break;
+
+    case ControlModes::CONTROL_TYPE::POSITION_DEGREES_ABSOLUTE:
+        return new DragonPercentOutputToCTREProAdapter(networkTableName,
```

Comment:
Shouldn't this be DragonPositionDegree or a similar class instead of DragonPercentOutput?

Reply:
Yes; thanks for catching this